### ISSUE TITLE
Tweaking the dragonScore()-requirement to play nicely with lizans and ascension

### ISSUE THREAD

**Issue by Stadler76** [Kitteh6660/Corruption-of-Champions-Mod#269]
**TL/DR: I'll come up with the first few PRs real soon.**

I wanted to become a (fully TFed) lizan with dragon wings and a snake-like tongue (looking closely at the small basilisk image you'll see that it has such a tongue) and realized, that this quite a pita especially, without debug mode, since drakes heart replaces your tongue and snake oil removes your wings.

After a bit more thinking a came up with the idea for some modifications to make it possible to lower the dragonScore()-requirement to >= 6 in CoC Revamp. To reach this I plan to do the following steps (Not necessarily in that order)

My approach to this
---
The dragonScore() requirement to be considered dragon-morph starts at `if (dragonScore() >= 7)`

1. ~~I'll alter dragonScore(), so that its clearer, that dragon horns add 2 points to your dragonScore() aka merging those two and replacing `dragonScore++` with `dragonScore += 2` => `if (dragonScore() >= 7)` **Not affected!**~~ **DONE and accepted**
2. ~~I'll file a PR so that predator arms with dragon claws add a point to your dragonScore() and with lizard claws, they'll add a point to your lizardScore(). => `if (dragonScore() >= 7)` **Not affected!**~~ **DONE and accepted**
3. ~~I'll file a PR to add TONGUE_SNAKE to lizan TF (No need for a new tongueType) => `if (dragonScore() >= 6)`~~ **DONE and accepted**
4. ~~I'll file a PR to implement SKIN_TYPE_DRACONIC (See below). => `if (dragonScore() >= 6)` **Won't lower it more!**~~ **DONE and accepted**
5. ~~I'll file a PR to implement my two new eyeTypes: `EYES_DRAGON` and `EYES_LIZARD`. (See below) These would add a point to dragonScore() respectively lizardScore(). => `if (dragonScore() >= 6)` **Not affected!**~~ **DONE and accepted**

Steps 2 and 5 are to add more dragonScore-points. Ok ... and since I'm eager to see my eyeTypes being added to CoC Revamp :P (See below)

With step 4 I'll introduce the PlayerHelper-class for the Player-class just like MutationsHelper is for Mutations and add more and more methods to it, since checking, if a PC has scaly skin without the method hasScales() would be a pita to implement.
Anyway: I'll do the work to wade through all skinType checks and fix them to use hasScales() if applicable.

Any objections @Kitteh6660? Of course I'll do the usual debugging and playtest several TFs to check that everything runs smoothly.

### Note:
After playtesting this locally with my Dragon Mod I came to the conclusion, that a dragonScore()-requirement of >= 6 is a good compromise between tweaking ascension and making it easier to be considered being a lizan. And I won't lower it to >= 4 so that lizans with some dragon-features (horns+dragonfire+wings) can still gain the stat-bonuses.

About adding the new skinType
---
I have added a new skinType to my patch: SKIN_TYPE_DRACONIC and I'll probably file a PR for it.
Reasons:

- I like to obtain lizan skin colors without needing to remove my scales first
- lizans with human skinColors are weird imho
- I prefer my dragon to have a non-human aka lizan skin color for the scales. A disambiguation between dragon scales and (normal) lizan scales makes that step easier.
- dragon TF lets your skin color unaffected and I don't intend to change that, since its an unique Ember only coloration.
- with more disambiguations between dragons and lizans it makes tweaking player.race() easier, although there should still be features dragons should share with lizans and probably other races
- Dragon scales have a different description (with shield-shaped scales) and I see them somewhat tougher/more effective.
- There may be more dragon subraces of dragon hybrids in the future.
- ~~I want to keep the score requirement to be considered being a dragon-morph low (its >= 7 atm). With that PR it could be lowered to >= 6 again.~~ Fixed with PR #270 and #271 
- ~~Upon ascension you'll be left with only 5 dragonscore points so you will not be considered having started your journey as a dragon morph.~~ Fixed with PR #270 and #271 
- ~~However: With the new eyeType EYES_DRAGON + lowering the score requirement down to >= 6 (+ the other tweaks I mentioned above) this would work again.~~ Fixed with PR #270 and #271 
- ~~Anyway: You can still reject that PR @Kitteh6660. I just wanted to state the reasons behind my PR and why I have added that to my dragon patch.~~ PR accepted and merged :)

About adding new eyeTypes
---
- I like to grant reptilic races (dragon and lizan/basilisk only for now) their typical vertically slitted eyes with a second set of eyelids.
- I will update Ember's appearance too, meantioning the second set of eyelids.
- lizans get EYES_LIZARD and and dragons will have EYES_DRAGON (See the code snippet below)
- both eyeType will grant a point toward dragonScore() respectively lizardScore() and since you'll keep your eyes upon ascension this would make it easier to ascend while being a dragon aka more ascension race-tweaking.
- I'm planning to work on EYES_BASILISK later in my mod, so see that as a seperate suggestion handled in a seperate PR:
 - Either with high affection from Benoit or after you layed a few eggs (10? 20?) for him, he'll give them as a present
 - I could even think about a recipe for that, but I think gaining a Basil Womb and laying eggs for him a few times is enough work for that.
 - I plan, that EYES_BASILISK will add the option to **Petrify** Imps and to place them as imp statues in from or on top of your wall, instead of just killing them. Not that would scare them off, seeing one of their fellows being doomed to be a stone statue.
 - While dragons will overwrite basil eyes lizans will keep them.
 - With basil eyes you'll gain the Basilisk Resistance Perk a bit faster.
 - Later, since this requires more balancing (especially the other races) I'll probably add a new combat ability called **Stare** which would reduce your opponents speed by a certain amount (basilisks will be immune, of course).
 - You still won't be able to petrify your opponent in combat cept Imps after defeating them.
 - However: I could possibly add a Petrify option with high corruption leading to a bad end after a few petrifications where you'll join Lethice's basilisk army.
- I'll probably open up a seperate issue for the new `eyeType`s
```as3
public static const EYES_LIZARD:int   = 3;
public static const EYES_DRAGON:int   = 4; // Slightly different description/TF and *maybe* in the future(!) grant different perks/combat abilities
public static const EYES_BASILISK:int = 5; // NYI! Example for now!!! Maybe later and probably with high affection from Benoit.
```

**Comment by Stadler76** [comment]
Steps 1 and 2 done with PR #270. Step 3 (snake-like tongue for lizans) will be next

**Comment by Stadler76** [comment]
PR #270 was accepted.
Step 3 done with PR #271 (snake-like tongue for lizans). Step 4 (draconic skinType for dragons only) will be next

**Comment by Stadler76** [comment]
The section "**About adding the new skinType**" has been updated.

I have another idea to make life easier with player.race() rather, than just tweaking score and moving things up and down, but this needs more testing and moving back and forth between dragon and lizan, while watching their scores changing. In short: its just an example and subject to change.
```as3
			if (lizardScore() >= 4)
			{
				race = "lizan";
				if (lizardScore() >= 8)
					return race; // high lizan score? always return lizan
			}
			if (dragonScore() >= 4)
			{
```

**Comment by Stadler76** [comment]
PR #271 was accepted
Step 4 done with PR #273  (new dragon scales skinType implemented)
Step 5 (new eyeTypes) will be the last one.
but before I start with Step 5 I plan to make changes and changeLimit a property, since it makes life easier, because I don't have to pass these values around anymore
additionally I found a couple of bugs related to driders:
- Hummanus doesn't remove drider lower body
- drider lowerBody doesn't appear in the player appearance tab (atm it requires isTaur() to be displayed, where it should require isDrider() instead.

**Comment by Stadler76** [comment]
Ok, PR #275 filed and accepted.
Next up will be the new eyeType's
And I decided to include TONGUE_SNAKE in lizardScore()

**Comment by Stadler76** [comment]
Ok, I just filed PR #278

**Comment by Stadler76** [comment]
**Step 5 of issue #269**
- I prefer to use the body part constants rather than their hardcoded numeric values, so I replaced some of them in dragonScore() and lizardScore()
- Tiny fix in emberTFs: "[...]<b>You have as many horns as a <del>lizan</del><ins>dragon</ins> can grow.</b>"
- new PlayerHelper-methods: hasDragonHorns() and hasReptileEyes()

**Comment by Stadler76** [comment]
With PR #278 being accepted and merged, I'll close this issue.
About the 'high lizan score' part, I mentioned above: I guess, I have to tweak this a little more.
